packages/starlight/integrations/heading-links.ts

```diff
@@ -56,7 +56,7 @@ export default function rehypeAutolinkHeadings(
 				{
 					type: 'element',
 					tagName: 'a',
-					properties: { class: 'sl-anchor-link', href: '#' + node.properties.id },
+					properties: { class: 'sl-anchor-link', href: '#' + String(node.properties.id) },
```

Comment:
Hah, I _guess_ the explicit coercion is better? Not something that really bothers me in the existing version of the code, but don’t mind too much either way.